downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/SBMLDumper.java

```diff
@@ -0,0 +1,300 @@
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Source;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.InvalidAttributeException;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+public class SBMLDumper {
+	// Initialize data structures used in module
+	private static HashMap<String,HashMap<String,ArrayList<String>>> reactions = new HashMap<String,HashMap<String,ArrayList<String>>>();
+	private static HashMap<String,String> compartments = new HashMap<String,String>();
+	private static HashMap<String,HashMap<String,String>> species = new HashMap<String,HashMap<String,String>>();
+	
+	// Retrieve all human ReactionlikeEvents
+	public static void execute(MySQLAdaptor dba) throws Exception {
+		GKInstance human = dba.fetchInstance(48887L);
+		Collection<GKInstance> reactionInstances = dba.fetchInstanceByAttribute(ReactomeJavaConstants.ReactionlikeEvent, ReactomeJavaConstants.species, "=", human);
+		int count = 0;
+		for (GKInstance reactionInst : reactionInstances) {
+			dissectEvent(reactionInst);
+			count++;
+			if (count == 50) {
+				break;
+			}
+		}
+		
+		generateSBMLFile();
+		validateSBMLFile();
+	}
+	
+	// Retrieve input/output/catalyst data for each reaction and update the reactions hashmap
+	public static void dissectEvent(GKInstance reactionInst) throws InvalidAttributeException, Exception {
+		
+		boolean validReaction = false;
+		for (GKInstance inputInst : (Collection<GKInstance>) reactionInst.getAttributeValuesList(ReactomeJavaConstants.input)) {
+			if (inputInst.getAttributeValue(ReactomeJavaConstants.compartment) != null) {
+				validReaction = true;
+			}
+		}
+		for (GKInstance outputInst : (Collection<GKInstance>) reactionInst.getAttributeValuesList(ReactomeJavaConstants.output)) {
+			if (outputInst.getAttributeValue(ReactomeJavaConstants.compartment) != null) {
+				validReaction = true;
+			}
+		}
+		
+		String id = getId(reactionInst);
+		
+		// Single name value added to ArrayList. This is needed to comply with the structure of the 'reactions' hashmap
+		if (validReaction) {
+			HashMap<String,ArrayList<String>> reactionsInfo = new HashMap<String,ArrayList<String>>();
+			ArrayList<String> name = new ArrayList<String>(Arrays.asList(reactionInst.getAttributeValue(ReactomeJavaConstants.name).toString()));
```

Comment:
Minor point -- could use `Collections.singleton()` in place of `Arrays.asList()` to show explicitly the list has one element